server/src/test/java/io/crate/analyze/AlterTableRerouteAnalyzerTest.java

```diff
@@ -201,13 +225,21 @@ public void test_promote_replica_shard_with_parameters() {
     }
 
     @Test
-    public void test_promote_replica_fails_if_unsupported_option_is_provided() {
+    public void testPromoteReplicaShardWithNullShardId() {
+        expectedException.expect(IllegalArgumentException.class);
```

Comment:
As the `ExpectedException` rule is deprecated since junit 4.13, we use `assertThrows` or our own `assertThrowsMatches` helper for new test cases instead.